### PR TITLE
Weather Function:

### PR DESCRIPTION
**Pull request by CodingRoy**
Mention the bot and say: Weather 'city' 'state/country'
Example:

> @[name-of-bot] Weather Amsterdam NL

### CODE REVIEW

**Comment by CodingRoy** [comment]
And don't forget to add the Wunderground API key of course ;) xD

File: src/index.js [hunk 1]
```diff
@@ -1,35 +1,70 @@
 var Botkit = require('botkit')
 var controller = Botkit.slackbot()
+var http = require('http');
+var weatherkey = 'Your wunderground key'; //wunderground api key
```
Comment (stephan281094) [inline src/index.js]: Please use the configuration variable from `.npmrc`:

``` js
process.env.WUNDERGROUND_API_KEY || process.env.npm_config_WUNDERGROUND_API_KEY
```

File: src/index.js [hunk 2]
```diff
@@ -1,35 +1,70 @@
 var Botkit = require('botkit')
 var controller = Botkit.slackbot()
+var http = require('http');
```
Comment (stephan281094) [inline src/index.js]: Please respect the standardjs code styling. Check linting errors by running `npm run lint`.

File: src/index.js [hunk 3]
```diff
@@ -1,35 +1,70 @@
...
+    response.on('end', function(){
+      var data = JSON.parse(body);
+      var days = data.forecast.simpleforecast.forecastday;
+      bot.reply(message, days[1].date.weekday +
+      ' high: ' + days[1].high.celsius +
+      ' low: ' + days[1].low.celsius +
+      ' condition: ' + days[1].conditions);
+      if(days[1].icon == 'clear'){
+        bot.reply(message, ':sunny:');
+      } else if(days[1].icon == 'partlycloudy' || days[1].icon == 'partlysunny'){
```
Comment (stephan281094) [inline src/index.js]: For readability, use a switch statement instead.

File: src/index.js [hunk 4]
```diff
@@ -1,35 +1,70 @@
...
-)
+})
 
 controller.hears('yo', ['ambient'], function (bot, message) {
   bot.reply(message, 'yo')
 })
+
+controller.hears("weather (.*) (.*)", ["mention", "direct_mention", "direct_message"], function(bot,message){
+  var city = message.match[1];
+  var state = message.match[2]; // state can also be a country but use states for the US!
```
Comment (stephan281094) [inline src/index.js]: Make the state argument optional by setting a default argument based on a configuration setting. E.g. `WUNDERGROUND_DEFAULT_STATE` inside `.npmrc`. Make sure to explain this in the readme.

File: src/index.js [hunk 5]
```diff
@@ -1,35 +1,70 @@
...
+  var url = '/api/' + weatherkey + '/forecast/q/state/city.json'
+  url = url.replace('state', state);
```
Comment (stephan281094) [inline src/index.js]: Instead of replacing both `state` and `city`, use string concatenation.

File: src/index.js [hunk 6]
```diff
@@ -1,35 +1,81 @@
...
-)
+})
 
 controller.hears('yo', ['ambient'], function (bot, message) {
   bot.reply(message, 'yo')
 })
+
+controller.hears('weather (.*)', ['mention', 'direct_mention', 'direct_message'], function (bot, message) {
+  var city = message.match[1] + '.json'
+  var state = process.env.WEATHER_DEFAULT_STATE || process.env.npm_config_WEATHER_DEFAULT_STATE  // state can also be a country but use states for the US!
```
Comment (CodingRoy) [inline src/index.js:35]: The state is still not optional, I can't find a way to make it optional atm.
Looks like something you can improve ;)

**Pull request merged**